RealmSwift/Realm.swift

```diff
@@ -475,10 +475,13 @@ public final class Realm {
     */
     public func addNotificationBlock(block: NotificationBlock) -> NotificationToken {
         return rlmRealm.addNotificationBlock { rlmNotification, _ in
-            if rlmNotification == NSNotification.Name.RLMRealmDidChange.rawValue {
-                block(notification: Notification.DidChange, realm: self)
-            } else if rlmNotification == NSNotification.Name.RLMRealmRefreshRequired.rawValue {
-                block(notification: Notification.RefreshRequired, realm: self)
+            switch rlmNotification {
+            case RLMNotification.DidChange:
+                block(notification: .DidChange, realm: self)
+            case RLMNotification.RefreshRequired:
+                block(notification: .RefreshRequired, realm: self)
+            default:
+                fatalError("Unhandled notification type: \(rlmNotification)")
```

Comment:
The only _functional_ change is we now crash when propagating an unknown notification type in Swift. We used to silently ignore such a condition. Either way, this should never happen in user code. Maybe I should change `fatalErrror` to `preconditionFailure` to make this more clear?